lib/temporal/workflow/context.rb

```diff
@@ -123,27 +124,43 @@ def execute_workflow(workflow_class, *input, **args)
         )
 
         target, cancelation_id = schedule_command(command)
-        future = Future.new(target, self, cancelation_id: cancelation_id)
+
+        # child_workflow_execution_future tracks when the workflow execution starts
+        child_workflow_execution_future = Future.new(target, self, cancelation_id: cancelation_id)
+
+        child_workflow_future = ChildWorkflowFuture.new(
+          target, 
+          self, 
+          cancelation_id: cancelation_id, 
+          child_workflow_execution_future: child_workflow_execution_future
+        )
+
+        has_started_child_workflow_execution = false
```

Comment:
This is unused, right? ^

Reply:
yes! sorry was using this inside the `error` handler to check if the future needed to be failed or not (then changed to how it is now with the current line)